hivemind/dht/__init__.py

```diff
@@ -105,10 +108,16 @@ def run(self) -> None:
 
             async def _run():
                 try:
+                    if hasattr(self, "_daemon_listen_maddr"):
+                        replicated_p2p = await P2P.replicate(self._daemon_listen_maddr)
+                    else:
+                        replicated_p2p = None
```

Comment:
```suggestion
                    if self._daemon_listen_maddr is not None:
                        replicated_p2p = await P2P.replicate(self._daemon_listen_maddr)
                    else:
                        replicated_p2p = None
```

It's better to avoid using `hasattr()` if possible (so no field names are a part of string constants, thus renames/type checking work well).